### PR TITLE
feat(docs): document AES block cypher

### PR DESCRIPTION
**Pull request by scriptandcompile**
second verse, same as the first!

### CODE REVIEW

File: src/ciphers/aes.rs
Comment (scriptandcompile) [inline src/ciphers/aes.rs:1]: I had meant to break each of these into their own commit, but I screwed up the patch add on the last one. Oopsie.

File: src/ciphers/aes.rs [hunk 1]
```diff
@@ -74,6 +143,17 @@ const INV_SBOX: [Byte; 256] = [
 ];
 
 #[rustfmt::skip]
```
Comment (alexfertel) [inline src/ciphers/aes.rs:145]: Let's put the attribute macro below the doc comments.

File: src/ciphers/aes.rs [hunk 2]
```diff
@@ -219,18 +299,71 @@ const GF_MUL_TABLE: [[Byte; 256]; 16] = [
...
+/// `key` - The AES key to use for encryption
+///
+/// # Returns
+///
+/// The encrypted text
+///
+/// # Panics
+///
+/// This function will not panic.
+///
```
Comment (alexfertel) [inline src/ciphers/aes.rs]: If a function doesn't panic, then this section can be skipped.

**Review by alexfertel — DISMISSED** [review]
LGTM! Only two small nits.

**Review by alexfertel — APPROVED** [review]
